preprocess_web/code/ravens_metadata_apps/preprocess_jobs/views.py

```diff
@@ -125,6 +125,38 @@ def view_api_retrieve_rows(request):
         return JobUtil.retrieve_rows_csv(request, job, **frm.cleaned_data)
 
 
+@csrf_exempt
+def variable_display_endpoint(request):
+    """ API endpoint to get JSON request for updating Preprocess file"""
+    if request.method != 'POST':
+        user_msg = dict(success=False,
+                            message='Please use a POST to access this endpoint')
+        return JsonResponse(user_msg)
+    # get json file from POST using view_helper
+    update_json = get_request_body_as_json(request)[1]
+    print("update Json : ",update_json)
+    if 'preprocess_id' in update_json:
+        preprocess_id = update_json['preprocess_id']
+        print(preprocess_id)
+        try:
+            job = PreprocessJob.objects.get(pk=preprocess_id)
+        except PreprocessJob.DoesNotExist:
+            result = 'job_id not found: %s' % preprocess_id
+            raise Http404('job_id not found: %s' % preprocess_id)
+
+        result = JobUtil.variable_display_job(job.get_preprocess_data(),update_json)
+        print('result ',result)
+
+    else:
+        result = dict(success=False,
+                      message='Preprocess_id not found !'
+                      )
```

Comment:
Need to return a JsonResponse with an error message here.  If not, it will go on to a success message